### ISSUE TITLE
[Enhancement]: Block-based checkout - update the Place order button

### ISSUE THREAD

**Issue by ralucaStan** [woothemes/woocommerce#48433]
This is a follow-up ticket for https://github.com/woocommerce/woocommerce/issues/46010. 
We ended up reverting the initial approach from https://github.com/woocommerce/woocommerce/pull/47083. 
The initial approach was introducing a `<price /> ` placeholder for the editor experience as seen in [this image](https://private-user-images.githubusercontent.com/3323310/328004301-d6a4fa79-d2a1-4939-b871-2460a8cf0c95.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxOTk2MzAsIm5iZiI6MTcxODE5OTMzMCwicGF0aCI6Ii8zMzIzMzEwLzMyODAwNDMwMS1kNmE0ZmE3OS1kMmExLTQ5MzktYjg3MS0yNDYwYThjZjBjOTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTJUMTMzNTMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGU1ZjZjYTg0MzA0NDEwNmM0NWZjMzczNTFiODVmNTRkYzU1ZWRjYzRjMmIwNmM1NWIxNjM0NDU5MDVhMGYyYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.XeCov4igcy_DjwLzP1X5NnMlGgP3KhlKH3Kwk9KRJtI)

Some of the reasons we are reconsidering the initial implementation:
- possible confusion around the `<price />` placeholder 
- the placeholder was not merchant friendly
- there are some concerns for the solution in an RTL env
- an opportunity to leverage the upcoming Bits API, possibly available in WordPress 6.6

-----

## Solution we want 
 Align with WooPay by adding a price to the purchase button. This is helpful especially when the sticky summary is not yet implemented

## Describe alternatives you've considered
Ongoing internal discussion about adding controls to toggle the display of price
![frame-3280](https://github.com/woocommerce/woocommerce/assets/1628454/eaaf4f8d-8aac-466d-ae57-8fd6b815ebb8)
pca54o-7Fa-p2#comment-6657
## Additional context

> [!IMPORTANT]
> We already have the `placeOrderButtonLabel` filter in place, which enables the modification of the "Place order" button label. We should update this filter to also allow merchants to optionally remove the total price from the button. Additionally, it is essential to test this implementation against the use of existing filters to ensure backwards compatibility.

**Comment by ralucaStan** [comment]
The team feels that the Bits API is still in an early phase, without a visual component, the markup for it isn't merchant-friendly. We decided to continue with the proposal to add a setting to Show the price on button.

**Comment by pmcpinto** [comment]
Raluca, thanks for creating the follow-up issue. @nikkivias @elizaan36 we can continue the discussion here.

As I mentioned in the comment, the toggle is a viable approach. My concern is that we end up with two different ways to edit the button content:
- Toggle to show/hide the price
- Inline edit to change the “place order” copy

I wonder if it would be better to focus on the inline edit flow and add the following instruction to the inspector when the place order button is selected: The <price/> will show the order total in the button. If it was deleted before, you can add <price/> to reintroduce it on the button. (the copy should be refined).

**Comment by elizaan36** [comment]
Thanks for the ping here! I would lean towards removing the inline editing capability for the price placeholder only. So that the toggle would be the only way to turn it on and off. Is it possible to keep inline editing for the "Place order" text and not the price?

The current placeholder `<price/>` would be confusing to the merchant. Could we use just an italic `_Price_` to indicate that the Order total will be there? Or the price of one product, such as `_$5_` for more of an accurate preview?

**Comment by ralucaStan** [comment]
I am unblocking this issue so that it can be picked up with the hope that getting started with it will help answer some of the open questions from the previous comments.

**Comment by opr** [comment]
Question for @nikkivias or @elizaan36. What should we do when the button text is long enough to run onto two lines, like this:

<img width="602" src="https://github.com/woocommerce/woocommerce/assets/5656702/91dc6f41-2853-4d11-b6b6-95f201cb7bbd" />

In the example above I've separated the text and price into two different elements, so they can be side-by-side. The problem with this approach is we don't know how wide the text is going to be so it could affect how it looks. Here's an example of when the screen (and therefore the button) is a little wider.

<img width="602" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/97985f1c-8534-42f3-a5b1-b94172d80c29" />

The alternative is to display these inline so the entire string takes up the whole button, like so:

<img width="602" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/69fb60a5-169a-49b4-bbee-8ac615bc6578">

I don't think any of these look particularly good unless the dimensions of the button and text match up well. Wondering if you could provide some guidance on what to do.

Thanks!

**Comment by elizaan36** [comment]
@opr As long as the price doesn't wrap I think it looks fine. Why is the price wrapping below the text in the 3rd example? If they are two separate elements could we ensure that each has a minimum width?

**Comment by opr** [comment]
Thanks for looking @elizaan36 !

> Why is the price wrapping below the text in the 3rd example?

It was an example of what it would look like if we treated the whole thing (text - separator - price) as one single element, so the whole thing would wrap onto two lines if it went longer than the button. Just to illustrate the scenario, I wasn't advocating we go with it.

> If they are two separate elements could we ensure that each has a minimum width?

~Yes, but I don't think this is going to solve the issue. The problem is when the text wraps onto two lines there's _additional_ space between the text and the separator, so it looks unbalanced, [this codepen has an example of what I mean](https://codepen.io/opr/pen/eYajQRv). I've added a black box around the text so you can see more clearly the actual space it's taking up.~

~Based on this answer https://stackoverflow.com/a/37413580/13516247 making sure the gap between the text, separator, and price is equal is not possible.~

I wrote all the above out, then went back to mess with the [codepen](https://codepen.io/opr/pen/eYajQRv) and I have something I think _might_ work. I'll try it out later!

**Comment by opr** [comment]
The [previous PR](https://github.com/woocommerce/woocommerce/pull/49252) was reverted due to some concerns around how the separator was implemented. It was implemented using CSS and we were worried that supporting this in future would limit the way we could change the button.

Darren suggested looking into the Block Style Variation API to control the separator. I recommend that next steps for this PR are looking into that API and validating whether it allows the separator to be customised in a more future-proof way.
